src/main/perf/LineFileDocs.java

```diff
@@ -884,42 +867,31 @@ private static final class BinaryBased extends LineFileDoc {
 
       final ByteBuffer blockByteText;
       private int nextId; // will have multiple doc in a same LFD so we'll determine id using a base
-      private int docCount; // and a count
+      private int remainingDocs; // and a count
 
       BinaryBased(ByteBuffer bytes, float[] vector, int idBase, int docCount) {
         super(vector);
         blockByteText = bytes;
         this.nextId = idBase;
-        this.docCount = docCount;
+        this.remainingDocs = docCount;
       }
 
       @Override
       int getNextId() {
-        if (docCount-- == 0) {
-          throw new IllegalStateException("Calling getId more than docCount");
+        if (remainingDocs-- == 0) {
```

Comment:
Paranoia: an we set `remainingDocs` back to 0 in this case?

This way if the truly insane caller tries again to `getNextId()` they will again get the `IllegalStateException`.

Reply:
Ah good catch, let me make the condition `<= 0` then?